packages/compiled-assets/src/index.ts

```diff
@@ -148,6 +154,14 @@ function compiledPath(type: 'scripts' | 'stylesheets', sourceFile: string): stri
   const sourceFilePath = `${type}/${sourceFile}`;
 
   if (options.dev) {
+    // To ensure that errors that would be raised in production are also raised
+    // in development, we'll check for the existence of the asset file on disk.
+    // This mirrors the production check of the file in the manifest: if a file
+    // exists on disk, it should be in the manifest.
+    if (!relativeSourcePaths?.find((p) => p === sourceFilePath)) {
+      throw new Error(`Unknown ${type} asset: ${sourceFile}`);
```

Comment:
Really nit: it's known that esbuild does not pick up new files added while the server is running, which has the (admittedly small) chance of causing a confusion for instructors. Do you think it's worth adding a quick check here to verify if the file exists and create a different error message (e.g., server needs to be restarted) if the file exists? Not a big deal if not, I don't think this is a common error.

Reply:
Instructors themselves will never encounter this as we only use esbuild for first-party code (that is, not course code).

Reply:
Sorry, by instructors I meant developers (the problem with filling both roles), but I see your point.